src/libappimage/desktop_integration/integrator/Integrator.cpp

```diff
@@ -258,20 +316,20 @@ namespace appimage {
                         if (relativeParentPath == defaultXdgDataDirPath)
                             relativeParentPath.clear();
                     }
-
-                    bf::path newPath = xdgDataHome / relativeParentPath / fileNameBuilder.str();
-                    return newPath;
+                    return relativeParentPath;
                 }
 
                 void deployMimeTypePackages() {
-                    auto mimeTypePackagesPaths = resourcesExtractor.getMimeTypePackagesPaths();
-                    std::map<std::string, std::string> mimeTypePackagesTargetPaths;
-
-                    // Generate deploy paths
-                    for (const auto& path: mimeTypePackagesPaths)
-                        mimeTypePackagesTargetPaths[path] = generateDeployPath(path).string();
-
-                    resourcesExtractor.extractTo(mimeTypePackagesTargetPaths);
+                    for (auto& itr: mimeInfoFiles) {
+                        MimeInfoEditor& editor = itr.second;
+                        editor.setDeployId(VENDOR_PREFIX + '_' + appImageId);
+                        boost::filesystem::path deployPath = generateDeployPath(itr.first);
+
+                        create_directories(deployPath.parent_path());
+                        std::ofstream out(deployPath.string());
+                        out << editor.edit();
```

Comment:
That method name `edit()` seems misleading.

Reply:
Indeed, rename to `getResult`